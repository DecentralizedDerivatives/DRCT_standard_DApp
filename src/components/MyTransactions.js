import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Table } from 'reactstrap';
import Loading from './Loading';
import { getUserTransactions } from '../actions/userActions';
import { SET_USER_TRANSACTIONS } from '../actions/types';
// Use named export for unconnected component for testing
export class MyTransactions extends Component {
  renderRows = () => {
    if (this.props.loading) {
      return <tr><td colSpan='12' style={{textAlign: 'center'}}><Loading /></td></tr>
    }
    if (this.props.userTransactions.length === 0) {
      return <tr><td colSpan='12' style={{textAlign: 'center'}}><h5>No Recent Events</h5></td></tr>
    }
    // console.log('Transactions', this.props.userTransactions)
    var rows = this.props.userTransactions.map((trade, index) => {
      const tradeTitle = trade.title;
      const tradeHash = trade.hash;

      return (
        <tr key={index}>
          <td>{tradeTitle}</td>
          <td>
            <a target="_blank" className="link__token-address"
              href={
                tradeHash.length > 50
                  ? `https://rinkeby.etherscan.io/tx/${tradeHash}`
                  : `https://rinkeby.etherscan.io/address/${tradeHash}`
              }
            >{tradeHash.substring(0, 14)}...
            </a>
          </td>
        </tr>
      );
    });
    return rows;
  };

  render() {
    return (
      <div className="table-container">
        <Table className="table table-hover table-striped table-responsive">
          <thead>
            <tr>
              <th colSpan='2'>My Transactions</th>
            </tr>
            <tr>
              <th>Transaction</th>
              <th>Transaction Hash</th>
            </tr>
          </thead>
          <tbody>{this.renderRows()}</tbody>
        </Table>
      </div>
    );
  }
}

MyTransactions.propTypes = {
  loading: PropTypes.bool.isRequired,
  userTransactions: PropTypes.array
};

const mapStateToProps = state => ({
  loading: state.status.fetchInProgress.includes(SET_USER_TRANSACTIONS),
  userTransactions: state.user.userTransactions
});

export default connect(
  mapStateToProps,
  { getUserTransactions }
)(MyTransactions);
