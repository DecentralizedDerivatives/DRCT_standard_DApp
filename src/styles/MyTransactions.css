.link__token-address: {
  color: #114770;
  text-decoration: none;
}
