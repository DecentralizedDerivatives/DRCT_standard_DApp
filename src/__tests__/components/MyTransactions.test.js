import MyTransactions from '../../components/MyTransactions';

describe('<MyTransactions />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <MyTransactions store={initFixtureStore()} />
      ).dive();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
