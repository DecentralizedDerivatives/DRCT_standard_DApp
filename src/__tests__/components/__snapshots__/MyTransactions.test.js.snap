// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyTransactions /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyTransactions
    getUserTransactions={[Function]}
    loading={false}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userTransactions={
      Array [
        Object {
          "hash": "0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb",
          "title": "ContractCreation",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="table-title"
        >
          My Transactions
        </div>,
        <table
          className="table table-hover table-striped"
        >
          <thead>
            <tr>
              <th
                style={
                  Object {
                    "width": "30%",
                  }
                }
              >
                Transaction
              </th>
              <th>
                Transaction Hash
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                ContractCreation
              </td>
              <td>
                <a
                  className="link__token-address"
                  href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                </a>
              </td>
            </tr>
          </tbody>
        </table>,
      ],
      "className": "wide-table-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Transactions",
          "className": "table-title",
        },
        "ref": null,
        "rendered": "My Transactions",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th
                  style={
                    Object {
                      "width": "30%",
                    }
                  }
                >
                  Transaction
                </th>
                <th>
                  Transaction Hash
                </th>
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td>
                  ContractCreation
                </td>
                <td>
                  <a
                    className="link__token-address"
                    href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                  </a>
                </td>
              </tr>
            </tbody>,
          ],
          "className": "table table-hover table-striped",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th
                  style={
                    Object {
                      "width": "30%",
                    }
                  }
                >
                  Transaction
                </th>
                <th>
                  Transaction Hash
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    style={
                      Object {
                        "width": "30%",
                      }
                    }
                  >
                    Transaction
                  </th>,
                  <th>
                    Transaction Hash
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Transaction",
                    "style": Object {
                      "width": "30%",
                    },
                  },
                  "ref": null,
                  "rendered": "Transaction",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Transaction Hash",
                  },
                  "ref": null,
                  "rendered": "Transaction Hash",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td>
                    ContractCreation
                  </td>
                  <td>
                    <a
                      className="link__token-address"
                      href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                    </a>
                  </td>
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      ContractCreation
                    </td>,
                    <td>
                      <a
                        className="link__token-address"
                        href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                      </a>
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "ContractCreation",
                    },
                    "ref": null,
                    "rendered": "ContractCreation",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        className="link__token-address"
                        href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                      </a>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb",
                        "className": "link__token-address",
                        "href": "https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb",
                        "rel": "noopener noreferrer",
                        "target": "_blank",
                      },
                      "ref": null,
                      "rendered": "0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb",
                      "type": "a",
                    },
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="table-title"
          >
            My Transactions
          </div>,
          <table
            className="table table-hover table-striped"
          >
            <thead>
              <tr>
                <th
                  style={
                    Object {
                      "width": "30%",
                    }
                  }
                >
                  Transaction
                </th>
                <th>
                  Transaction Hash
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  ContractCreation
                </td>
                <td>
                  <a
                    className="link__token-address"
                    href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                  </a>
                </td>
              </tr>
            </tbody>
          </table>,
        ],
        "className": "wide-table-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Transactions",
            "className": "table-title",
          },
          "ref": null,
          "rendered": "My Transactions",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th
                    style={
                      Object {
                        "width": "30%",
                      }
                    }
                  >
                    Transaction
                  </th>
                  <th>
                    Transaction Hash
                  </th>
                </tr>
              </thead>,
              <tbody>
                <tr>
                  <td>
                    ContractCreation
                  </td>
                  <td>
                    <a
                      className="link__token-address"
                      href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                    </a>
                  </td>
                </tr>
              </tbody>,
            ],
            "className": "table table-hover table-striped",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th
                    style={
                      Object {
                        "width": "30%",
                      }
                    }
                  >
                    Transaction
                  </th>
                  <th>
                    Transaction Hash
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th
                      style={
                        Object {
                          "width": "30%",
                        }
                      }
                    >
                      Transaction
                    </th>,
                    <th>
                      Transaction Hash
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Transaction",
                      "style": Object {
                        "width": "30%",
                      },
                    },
                    "ref": null,
                    "rendered": "Transaction",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Transaction Hash",
                    },
                    "ref": null,
                    "rendered": "Transaction Hash",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tr>
                    <td>
                      ContractCreation
                    </td>
                    <td>
                      <a
                        className="link__token-address"
                        href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                      </a>
                    </td>
                  </tr>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td>
                        ContractCreation
                      </td>,
                      <td>
                        <a
                          className="link__token-address"
                          href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                        </a>
                      </td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "ContractCreation",
                      },
                      "ref": null,
                      "rendered": "ContractCreation",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          className="link__token-address"
                          href="https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb
                        </a>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb",
                          "className": "link__token-address",
                          "href": "https://rinkeby.etherscan.io/tx/0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb",
                          "rel": "noopener noreferrer",
                          "target": "_blank",
                        },
                        "ref": null,
                        "rendered": "0x7e1bf33a238354bebfd59f2bbd88a99e75c0b75d9aaa5a2e4b4225e1e51369fb",
                        "type": "a",
                      },
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
              ],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
